ipa-core/src/protocol/basics/reveal.rs

```diff
@@ -182,13 +148,14 @@ impl<'a, F: ExtendableField> Reveal<UpgradedMaliciousContext<'a, F>, 1> for Mali
 
         // Send share to helpers to the right and left
         // send except to left_out
-        if ctx.role().peer(Direction::Left) != left_out {
-            left_sender.send(record_id, right).await?;
-        }
-        if ctx.role().peer(Direction::Right) != left_out {
-            right_sender.send(record_id, left).await?;
-        }
-        if ctx.role() == left_out {
+        let send_left_fut = (Some(ctx.role().peer(Direction::Left)) != left_out)
+            .then(|| left_sender.send(record_id, right));
+        let send_right_fut = (Some(ctx.role().peer(Direction::Right)) != left_out)
+            .then(|| right_sender.send(record_id, left));
+        ctx.parallel_join(send_left_fut.into_iter().chain(send_right_fut))
```

Comment:
A parallel join involves a bunch of machinery.  Is there a cheaper way to join two `Option<impl Future>` ?

Reply:
What I really wanted here is the native `try_join_all`. To avoid the parallel join overhead without trying to use the disallowed method, I added a `MaybeFuture` helper so that we can treat each of the left and right actions as futures whether or not they're actually sending.